common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -631,4 +656,12 @@ private static void setCorrelationIdOnResult(@NonNull final CommandResult comman
         }
     }
 
+
+    private static void performRefresh(BaseCommand command) {
+        SilentTokenCommandParameters parameters = ((SilentTokenCommandParameters) command.getParameters()).toBuilder().forceRefresh(true).build();
```

Comment:
Seems like we are just leveraging the `forceRefresh` here to execute refreshes that are triggered based on the presence of `refresh_in` returned from server. How do we distinguish this in the logs? i.e. How do we distinguish inside the logs when this command is executing that this wasn't triggered by some flow executed by the customer but rather this is just us doing force refreshes to renew tokens ahead of time?

Reply:
The CommandResult will return REFRESH when a refresh is needed...now this will submit another command to the dispatcher with force refresh true....I'm asking that for this command how do we distinguish that this is NOT a regular force refresh but rather a force refresh executed by the library because a refresh was needed.....Inside the logs all these requests are being executed in parallel...we don't know which command spun up another one...we either need to determine that...or internally when the refresh is happening inside the controller then have some mechanism to make the distinction there....

Ideally you want to print in the logs and specify that a specific command spun up this other command to do a refresh and print the correlation id of both so we can do the association